layers/data_service_layer/python/data_service.py

```diff
@@ -24,4 +24,55 @@ def get_markers(self) -> List[dict]:
         :return: A list of markers.
         :raises NotImplementedError: Indicates the method is not yet implemented.
         """
-        raise NotImplementedError("This method is not implemented yet.")
+        try: 
+            response = self.table.scan()
+            markers = response.get('Items', [])
+            return markers
+        except Exception as e:
+            logger.error(f"Error retrieving markers")
+
+    def add_marker(self, markerId, name, longitude, latitude, imgurl, tags):
+        """
+        Add a new marker to the DynamoDB table.
+
+        :param markerId: Unique identifier for the location.
+        :param Name: Name of the location.
+        :param Longitude: Longitude coordinate of the location.
+        :param Latitude: Latitude coordinate of the location.
+        :param ImgUrl: URL to an image of the location.
+        :param Tags: Tags associated with the location.
+        :return: The response from DynamoDB.
+        """
+        response = self.table.put_item(
```

Comment:
@PCBush instead of recreating an Item here, you could do

`.put_item(locationMarker.to_json())`